components/tidb_query/src/expr/builtin_math.rs

```diff
@@ -461,22 +442,12 @@ impl ScalarFunc {
     }
 }
 
-fn get_rand(arg: Option<u64>) -> XorShiftRng {
-    let seed = match arg {
-        Some(v) => v,
-        None => {
-            let current_time = time::get_time();
-            let nsec = current_time.nsec as u64;
-            let sec = (current_time.sec * 1000000000) as u64;
-            sec + nsec
-        }
-    };
-    SeedableRng::seed_from_u64(seed)
+thread_local! {
+   static MYSQL_RNG: RefCell<MySQLRng> = RefCell::new(MySQLRng::new())
```

Comment:
It would be better to wrap this thread local in a module, so that:

- There is no duplicate thread local for vec and non-vec version.
- The call site can be cleaner.

Reply:
I think it already simple enough, we can just keep it.